infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -300,6 +311,8 @@ private void bootstrapLogReplicationService() {
         // Unblock server initialization & register to Log Replication Lock, to attempt lock / leadership acquisition
         serverCallback.complete(interClusterReplicationService);
 
+        logReplicationEventListener = new LogReplicationEventListener(this);
```

Comment:
Is it worth to subscribe every node to listen to the enforce event? if anyways only the leader would take action... in which case maybe it would be better to subscribe onLockAcquire? and unsubscribe onLockRelease... anyways a node that is not a leader even if it listens it won't do anything...

Reply:
I thought about that but I don't see the point to complicate the logic.